osquery/tables/networking/posix/nslookup.cpp

```diff
@@ -0,0 +1,124 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#include <arpa/inet.h>
+#include <netdb.h>
+
+#include <osquery/core.h>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+
+namespace osquery {
+namespace tables {
+
+QueryData genLookup(QueryContext& context) {
+  QueryData results;
+
+  if (context.constraints.count("address") > 0 &&
+      context.constraints.at("address").exists(EQUALS)) {
```

Comment:
I think you can make a minor optimization here by avoiding some extra lookups...
```c++
auto addrs = context.constraints.find("address");
if (addrs != context.contraints.end()  &&
    addrs->second.exists(EQUALS)) {
    ...
    for (const auto& ip_str : addrs.getAll(EQUALS)) {
      ...
    }
```